Source/WebCore/html/HTMLElement.cpp

```diff
@@ -130,7 +130,9 @@ String HTMLElement::nodeName() const
 static inline CSSValueID unicodeBidiAttributeForDirAuto(HTMLElement& element)
 {
     ASSERT(!element.hasTagName(bdoTag));
-    if (element.hasTagName(preTag) || element.hasTagName(textareaTag))
+    ASSERT(!element.hasTagName(preTag));
+    ASSERT(!element.hasTagName(textareaTag));
+    if (RefPtr input = dynamicDowncast<HTMLInputElement>(element); input && (input->isTelephoneField() || input->isEmailField() || input->isSearchField() || input->isURLField() || input->isTelephoneField()))
```

Comment:
NOTE TO Myself - `isTelephoneField` is twice in this, so after review and green EWS, I will remove it before merging.

Reply:
Fixed. :-)